### PR TITLE
add GetAllIbaDashboards to the TwoStageL3ClosClient and tests

### PR DESCRIPTION
**Pull request by rajagopalans**

### CODE REVIEW

File: apstra/api_iba_dashboards_test.go [hunk 1]
```diff
@@ -114,13 +115,26 @@ func TestCreateReadUpdateDeleteIbaDashboards(t *testing.T) {
 			IbaWidgetGrid: [][]ObjectId{{widgetAId, widgetBId}, {widgetAId, widgetBId}},
 		}
 
-		t.Log("Test Create Dashboard")
+		ds, err := bpClient.GetAllIbaDashboards(ctx)
+		l := len(ds)
+		if len(ds) != 0 {
+			t.Fatalf("Expected no dashboards. got %d", l)
+		}
```
Comment (chrismarget-j) [inline apstra/api_iba_dashboards_test.go:122]: Needs error check.

Variable `l` is set to `len(ds)`, and then `len(ds)` is called immediately after. Should choose between using `l` and `len(ds)`. I'd probably eliminate `l` and just invoke `len()` a second time in the problem case.

File: apstra/api_iba_dashboards_test.go [hunk 2]
```diff
@@ -114,13 +115,26 @@ func TestCreateReadUpdateDeleteIbaDashboards(t *testing.T) {
...
 			t.Fatal(err)
 		}
 
+		ds, err = bpClient.GetAllIbaDashboards(ctx)
+		l = len(ds)
+		t.Logf("Found %d dashboards", l)
+		if len(ds) != 1 {
+			t.Fatalf("Expected 1 dashboards. got %d", l)
+		}
+
```
Comment (chrismarget-j) [inline apstra/api_iba_dashboards_test.go:137]: ditto.

File: apstra/two_stage_l3_clos_client.go [hunk 3]
```diff
@@ -645,6 +645,25 @@ func (o *TwoStageL3ClosClient) GetIbaWidget(ctx context.Context, id ObjectId) (*
...
+	}
+
+	dashes := make([]IbaDashboard, len(rawDashes))
+	for i, w := range rawDashes {
+		pw, err := w.polish()
+		dashes[i] = *pw
+		if err != nil {
+			return dashes, err
+		}
+	}
```
Comment (chrismarget-j) [inline apstra/two_stage_l3_clos_client.go:662]: Please move err check to immediately after the error is generated.

Please return `nil` in the error case.

**Review by chrismarget-j — CHANGES_REQUESTED** [review]

File: apstra/api_iba_dashboards_test.go (hunk 1 above)
Comment (chrismarget-j) [reply apstra/api_iba_dashboards_test.go:122]: un-checked error is above. It didn't quite make the context that GitHub has decided to share here in the conversation pane.

Comment (rajagopalans) [reply apstra/api_iba_dashboards_test.go:122]: Man! I was going to replace all lens with l. Got distracted before I finished the job :-)

**Review by chrismarget-j — APPROVED** [review]

**Pull request merged**